packages/ui/src/multiplying-architecture/KaotoEditorFactory.ts

```diff
@@ -1,24 +1,39 @@
-import { KaotoEditorApp } from './KaotoEditorApp';
 import {
   Editor,
   EditorFactory,
   EditorInitArgs,
   KogitoEditorEnvelopeContextType,
 } from '@kie-tools-core/editor/dist/api';
-import { KaotoEditorChannelApi } from './KaotoEditorChannelApi';
+import { DefaultSettingsAdapter } from '../models';
 import { CatalogSchemaLoader, isDefined } from '../utils';
+import { KaotoEditorApp } from './KaotoEditorApp';
+import { KaotoEditorChannelApi } from './KaotoEditorChannelApi';
 
 export class KaotoEditorFactory implements EditorFactory<Editor, KaotoEditorChannelApi> {
   public async createEditor(
     envelopeContext: KogitoEditorEnvelopeContextType<KaotoEditorChannelApi>,
     initArgs: EditorInitArgs,
   ): Promise<Editor> {
-    let catalogUrl = await envelopeContext.channelApi.requests.getCatalogURL();
+    const settings = await envelopeContext.channelApi.requests.getVSCodeKaotoSettings();
+    const settingsAdapter = new DefaultSettingsAdapter(settings);
+    this.updateCatalogUrl(settingsAdapter, initArgs);
+
+    return Promise.resolve(new KaotoEditorApp(envelopeContext, initArgs, settingsAdapter));
```

Comment:
@apupier, it seems we can't pass objects with functions through the `channelApi`, I was trying to pass the `SettingsAdapter` directly but the methods were stripped from the class, so I assume that at some point, the message gets serialized into string, removing the functions and methods.

Reply:
yes, it is also my understanding